content/en/docs/installing-on-kubernetes/on-prem-kubernetes/install-ks-on-linux-airgapped.md

```diff
@@ -1,10 +1,409 @@
+   ```
+
+   ```bash
+   openssl req \
+   -newkey rsa:4096 -nodes -sha256 -keyout certs/domain.key \
+   -x509 -days 36500 -out certs/domain.crt
+   ```
+
+2. Make sure you specify a domain name in the field `Common Name` when you are generating your own certificate. For instance, the field is set to `dockerhub.kubekey.local` in this example. 
+
+   ![self-signed-cert](/images/docs/installing-on-linux/introduction/air-gapped-installation/self-signed-cert.jpg)
+
+### Start Docker Registry
+
+Run the following commands to start the Docker registry:
+
+```
+docker run -d \
+  --restart=always \
+  --name registry \
+  -v "$(pwd)"/certs:/certs \
+  -v /mnt/registry:/var/lib/registry \
+  -e REGISTRY_HTTP_ADDR=0.0.0.0:443 \
+  -e REGISTRY_HTTP_TLS_CERTIFICATE=/certs/domain.crt \
+  -e REGISTRY_HTTP_TLS_KEY=/certs/domain.key \
+  -p 443:443 \
+  registry:2
+```
+
+{{< notice note >}}
+
+Docker uses `/var/lib/docker` as the default directory where all Docker related files, including images, are stored. It is recommended you add additional storage volumes with at least **100G** mounted to `/var/lib/docker` and `/mnt/registry` respectively. See [fdisk](https://www.computerhope.com/unix/fdisk.htm) command for reference.
+
+{{</ notice >}}
+
+### Configure Registry
+
+1. Add an entry to `/etc/hosts` to map the hostname (i.e. the registry domain name; in this case, it is `dockerhub.kubekey.local`) to the private IP address of your machine as below.
+
+   ```bash
+   # docker registry
+   192.168.0.2 dockerhub.kubekey.local
+   ```
+
+2. Execute the following commands to copy the certificate to a specified directory and make Docker trust it.
+
+   ```bash
+   mkdir -p  /etc/docker/certs.d/dockerhub.kubekey.local
+   ```
+
+   ```bash
+   cp certs/domain.crt  /etc/docker/certs.d/dockerhub.kubesphere.local/ca.crt
+   ```
+
+   {{< notice note >}}
+
+   The path of the certificate is related to the domain name. When you copy the path, use your actual domain name if it is different from the one set above.
+
+   {{</ notice >}} 
+
+3. To verify whether the private registry is effective, you can copy an image to your local machine first, and use `docker push` and `docker pull` to test it.
+
+## Step 2: Prepare Installation Images
+
+As you install KubeSphere in an air-gapped environment, you need to prepare an image package containing all the necessary images in advance.
+
+1. Download the image list file `images-list.txt` from a machine that has access to the Internet through the following command:
+
+   ```bash
+   curl -L -O https://github.com/kubesphere/ks-installer/releases/download/v3.0.0/images-list.txt
+   ```
+
+   {{< notice note >}}
+
+   This file lists images under `##+modulename` based on different modules. You can add your own images to this file following the same rule. To view the complete file, see [Appendix](../install-ks-on-linux-airgapped/#image-list-of-kubesphere-v300).
+
+   {{</ notice >}} 
+
+2. Download `images-manager.sh`. 
```

Comment:
This should be changed to `offline-installation-tool.sh`.

Reply:
Changed.